### PR TITLE
Docs: Storybook SWC builder updates

### PR DESCRIPTION
**Pull request by jonniebigodes**
## What I did

With this pull request the documentation was updated to factor in the upcoming SWC builder, which resulted in:
- Updating the existing babel documentation
- Set up a dedicated API page
- Creating snippets
- Fix misc links, outdated references and sidebar organization
### Testing

1. Follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `docs_swc_flag`.
2. Open the relevant documentation and verify the feature changes.

@valentinpalkovic, when you have a moment, can you check on this and let me know of any feedback you may have? Looking forward to hearing from you. Thanks in advance.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [X] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/configure/compilers.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+title: 'Compiler support'
+---
+
+Javascript compilers are essential in optimizing and transforming code, enhancing performance, and ensuring compatibility across different environments. Storybook provides support for the leading compilers, ensuring lightning-fast build time and execution with [SWC](https://swc.rs/) or leveraging [Babel](https://babeljs.io/) with its extensive ecosystem of plugins and presets to allow you to use the latest features of the ecosystem with minimal configuration required.
```
Comment (valentinpalkovic) [inline docs/configure/compilers.md:5]: We should make it clear in the very beginning that this page is just about configuring compilers for Webpack5-based projects. This whole page doesn't apply for vite Projects for example.

File: docs/configure/compilers.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+## SWC
+
+SWC is a fast, highly extensible tool for compiling and bundling modern JavaScript applications. Powered by [Rust](https://www.rust-lang.org/), it improves performance and reduces build times. Storybook includes a built-in integration with SWC, allowing zero-configuration setup and built-in types for APIs. If you've initialized Storybook in a Webpack-based project with any of the supported [frameworks](./frameworks.md), except Angular, it will automatically use SWC as its default, providing you with faster loading time. However, if you're upgrading from a previous version of Storybook, you may need to opt-in to use SWC by adjusting your Storybook configuration file (i.e., `.storybook/main.js|ts`) as follows:
```
Comment (valentinpalkovic) [inline docs/configure/compilers.md:9]: Except Angular, and Next.js. The SWC support for Next.js is experimental and therefore users have to opt in for SWC in Next.js

**Review by valentinpalkovic — COMMENTED** [review]
The rest LGTM! Well done! :)

File: docs/snippets/common/main-config-swc-jsx-transform.ts.mdx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        useSWC: true, // This flag is automatically set by Storybook for all new Webpack5 projects (except Angular) in Storybook 7.6
+      },
+    },
+  },
+  swc: (config, options) => ({
+    jsc: {
+      transform: {
+        runtime: 'automatic',
+      },
+    },
```
Comment (valentinpalkovic) [inline docs/snippets/common/main-config-swc-jsx-transform.ts.mdx:19]: Sorry, I gave you the wrong code snippet. It has to be:

```suggestion
    jsc: {
      transform: {
        react: {
          runtime: 'automatic'
        }
      }
    }
```

File: docs/snippets/common/main-config-swc-jsx-transform.js.mdx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        useSWC: true, // This flag is automatically set by Storybook for all new Webpack5 projects (except Angular) in Storybook 7.6
+      },
+    },
+  },
+  swc: (config, options) => ({
+    jsc: {
+      transform: {
+        runtime: 'automatic',
+      },
+    },
```
Comment (valentinpalkovic) [inline docs/snippets/common/main-config-swc-jsx-transform.js.mdx:19]: Sorry, I gave you the wrong code snippet. It has to be:

```suggestion
    jsc: {
      transform: {
        react: {
          runtime: 'automatic'
        }
      }
    }
```

File: docs/snippets/common/main-config-swc-jsx-transform.ts.mdx (hunk 3 above)
Comment (jonniebigodes) [reply docs/snippets/common/main-config-swc-jsx-transform.ts.mdx]: No problem whatsoever. Thanks for clarifying it. I've updated them with the latest commit.

**Review by valentinpalkovic — APPROVED** [review]

**Pull request merged**